sycl/source/handler.cpp

```diff
@@ -837,11 +860,25 @@ void handler::extractArgsAndReqsFromLambda(
 // Calling methods of kernel_impl requires knowledge of class layout.
 // As this is impossible in header, there's a function that calls necessary
 // method inside the library and returns the result.
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+detail::string handler::getKernelName() {
+  detail::string Str;
+  std::string Fname = MKernel->get_info<info::kernel::function_name>();
+  Str.allocate(Fname.length() + 1);
+  Str.unmarshall(Fname);
+  return Str;
+}
+#else
 std::string handler::getKernelName() {
   return MKernel->get_info<info::kernel::function_name>();
 }
+#endif
 
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+void handler::verifyUsedKernelBundleInternal(detail::string_view &KernelName) {
```

Comment:
```suggestion
void handler::verifyUsedKernelBundleInternal(detail::string_view KernelName) {
```
Just a pointer

Reply:
I am not sure if I understand this correctly.
Pass-by-value will create a copy which is heavier than pass-by-reference.

Reply:
Your class has a single member pointer, so copying is cheap and you'd save an extra indirection on each access.

Reply:
Oops. Right, now I have split two original data members (str and ret_str) into a single member in each class.
So, it is equivalent in sizes, but the reference parameter causes an indirection to the original object.
Thanks for explanation.